src/globalNotification/globalNotificationProvider.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { useContext, useState } from 'react';
+import GlobalNotification, {
+  Props as NotificationProps,
+} from './globalNotification';
+
+interface GlobalNotificationContextInterface {
+  notifications: NotificationProps[];
+  render: () => JSX.Element[] | null;
+  pushNotification: (notificationProps: NotificationProps) => void;
+  popNotification: (id?: string) => void;
+}
+
+interface GlobalNotificationProviderInterface {
+  children: JSX.Element;
+}
+
+const defaultContext: GlobalNotificationContextInterface = {
+  notifications: [],
+  render: () => null,
+  pushNotification: (notificationProps) => notificationProps,
+  popNotification: () => null,
+};
+
+export const GlobalNotificationCtx =
+  React.createContext<GlobalNotificationContextInterface>(defaultContext);
+
+const GlobalNotificationProvider = (
+  props: GlobalNotificationProviderInterface
+): JSX.Element => {
+  const [notifications, setNotifications] = useState<NotificationProps[]>([]);
+
+  const render = () =>
+    notifications.map((n) => (
+      <GlobalNotification
+        key={n.id}
+        id={n.id}
+        body={n.body}
+        type={n.type}
+        icon={n.icon}
```

Comment:
Not incorrect, though with the current props at least all this could just be shortened to `<GlobalNotification key={n.id} {...n} />`. (At present, the label isn't passed on here, but I believe it should.)